bundler/spec/bundler/fetcher/downloader_spec.rb

```diff
@@ -83,6 +83,11 @@
           /Authentication is required for www.uri-to-fetch.com/)
       end
 
+      it "should raise a Bundler::Fetcher::AuthenticationRequiredError with advices" do
+        expect { subject.fetch(uri, options, counter) }.to raise_error(Bundler::Fetcher::AuthenticationRequiredError,
+          /`bundle config set --global www\.uri-to-fetch\.com username:password`.*`BUNDLE_WWW__URI-TO-FETCH__COM`/m)
```

Comment:
The `-` characters in `BUNDLE_WWW__URI-TO-FETCH__COM` aren't [valid in variable names](https://www.tutorialspoint.com/unix/unix-using-variables.htm#:~:text=Variable%20Names,have%20their%20names%20in%20UPPERCASE.), although I'm not sure whether this matters for the test

Reply:
Spec failure was not related, but that's a good point though!

We should either fix that issue right now, or open a separate ticket about it and avoid the ENV variable recommendation in that particular case.

Reply:
According to https://stackoverflow.com/questions/3523028/valid-characters-of-a-hostname, only letters, numbers, and hyphens are valid within a hostname component, so I guess we can add the `___` (triple underscore) <-> `-` (hyphen) to this translation and fix the issue completely.

Reply:
I proposed a fix at https://github.com/rubygems/rubygems/pull/4571.